qcodes/instrument_drivers/AlazarTech/ATS.py

```diff
@@ -445,6 +445,36 @@ def acquire(self, mode=None, samples_per_record=None,
         samples_per_record = self.samples_per_record.raw_value
         records_per_buffer = self.records_per_buffer.raw_value
 
+        # bits per sample
+        _, bits_per_sample = self.api.get_channel_info_(self._handle)
+        # Whole bytes per sample
+        bytes_per_sample = (bits_per_sample + 7) // 8
+        # bytes per record
+        bytes_per_record = bytes_per_sample * samples_per_record
+
+        # channels
+        channels_binrep = self.channel_selection.raw_value
+        number_of_channels = self.get_num_channels(channels_binrep)
+
+        # bytes per buffer
+        bytes_per_buffer = (bytes_per_record *
+                            records_per_buffer * number_of_channels)
+
+        sample_type = ctypes.c_uint16 if bytes_per_sample > 1 else ctypes.c_uint8
+
+        max_buffer_size = 64*1024*1024
+        # 64 MB see docs of AlazarBeforeAsyncRead
+        # http://www.alazartech.com/Support/Download%20Files/ATS-SDK-Guide-7.2.3.pdf#section*.110
+
+        requested_buffer_size = (bits_per_sample * samples_per_record *
```

Comment:
I find it hard to understand why `number_of_channels` of channels is not involved in this formula. The respective formula in SDK pdf also does not contain `number_of_channels`. Why is that? Are on-board-memory-buffers considered per channel, and host-memory-buffers not?

Reply:
This buffer size is only a fraction of the total buffer size (1/64) in this case. I don't think that the limitation comes from filling the internal memory but from some other implementation detail that they don't reveal to use

Reply:
> This buffer size is only a fraction of the total buffer size (1/64) in this case.

Sorry, i'm getting more confused. What did you mean with "this buffer size" and "total buffer size" and where does "1/64" comes from? and how does this relate to taking `number_of_channels` into account or not?

> ... from some other implementation detail ...

i agree with you, i have the same feeling.

Reply:
Sorry should have said total onboard memory. I mean given that we don't know the reason for this limitation (other than it's for sure not filling all the memory cf the card which would for sure depend on number of channels) all questions about this are speculations. This limitation might just be that the whatever fpga logic that captures data from a single channel only has a memory address space large enough to address 64 MB and there is a matching address space for the next channel. We really don't know at this stage.

Reply:
ok, thanks for explaining! I hope AlazarTech come back with useful answers :)